shell/platform/embedder/embedder.cc

```diff
@@ -532,6 +533,70 @@ struct _FlutterPlatformMessageResponseHandle {
   fml::RefPtr<flutter::PlatformMessage> message;
 };
 
+struct LoadedELFDeleter {
+  void operator()(Dart_LoadedElf* elf) { ::Dart_UnloadELF(elf); }
+};
+
+using UniqueLoadedELF = std::unique_ptr<Dart_LoadedElf, LoadedELFDeleter>;
+
+struct _FlutterEngineAOTData {
+  UniqueLoadedELF loaded_elf = nullptr;
+  const uint8_t* vm_snapshot_data = nullptr;
+  const uint8_t* vm_snapshot_instrs = nullptr;
+  const uint8_t* vm_isolate_data = nullptr;
+  const uint8_t* vm_isolate_instrs = nullptr;
+};
+
+FlutterEngineResult FlutterEngineCreateAOTData(
+    const FlutterEngineAOTDataSource* source,
+    FlutterEngineAOTData* data_out) {
+  switch (source->type) {
+    case kFlutterEngineAOTDataSourceTypeElfPath: {
+      if (!source->elf_path || !fml::IsFile(source->elf_path)) {
+        return LOG_EMBEDDER_ERROR(kInvalidArguments,
+                                  "Invalid ELF path specified.");
+      }
+
+      auto aot_data = new _FlutterEngineAOTData();
+      const char* error = nullptr;
+
+      auto loaded_elf = Dart_LoadELF(
```

Comment:
Style guide strongly discourages using `auto` when the type isn't obvious from the right hand side, so this would be better as an explicit type.